tests/main/retry-network/task.yaml

```
@@ -24,6 +24,15 @@ debug: |
 execute: |
     echo "Add totally unconnected network namespace"
     ip netns add testns
+
+    for _ in $(seq 10); do
+        nsenter --net=/var/run/netns/testns ./detect-retry http://www.ubuntu.com > output.txt
+        if MATCH "NoNetwork: true" < output.txt; then
+            break
+        fi
+        sleep 1
+    done
```

A minor nitpick, but I think this would be a bit nicer with retry-tool and tee, i.e.

```suggestion
    retry-tool -n 10 --wait 1 bash -c 'nsenter --net=/var/run/netns/testns ./detect-retry http://www.ubuntu.com | tee output.txt | MATCH "NoNetwork: true"'
```